### ISSUE TITLE
use NodeJS Buffer problem

### ISSUE THREAD

**Issue by gn0262487838** [zhangyuang/node-ffi-rs#52]
## Current ffi-rs version

version: "ffi-rs": "^1.0.78"

other module:
1. "iconv": "^3.0.1",

## NodeJS version

version: 18.18.2 32bit

## OS info

OS: Windows 11 home 23H2 64bit

## C++ sample code

```cpp
#include <cstring>

extern "C" __declspec(dllexport) int myTest(char *buffer, int *length)
{
    // Simulate sending Chinese (encoding: Big5) back to nodeJS
    const char *chineseStr = "\xA4\xA4\xA4\xE5"; // 中文
    int chineseLen = 4;

    if (*length < chineseLen)
    {
        return -1;
    }

    memcpy(buffer, chineseStr, chineseLen);
    *length = chineseLen;
    return 0;
}

// command: gcc -shared -o myTest.dll myTest.cpp -Wl,--out-implib,libmydll.a
```

## NodeJS sample code:

```typescript
import { DataType, load, open } from 'ffi-rs';
import { Iconv } from 'iconv';
const iconv = new Iconv('Big5', 'UTF-8');

const DEFAULT_DLL_KEY = 'myTest';

open({
    library: DEFAULT_DLL_KEY,
    path: './myTest.dll',
});

function testFun() {
    let errorCode: any;
    try {
        const bufferSize = 4;
        const pBuffer: Buffer = Buffer.alloc(bufferSize);
        const pBufferLen: Buffer = Buffer.alloc(4);
        pBufferLen.writeInt32LE(bufferSize);
        errorCode = load({
            library: DEFAULT_DLL_KEY,
            funcName: 'myTest',
            retType: DataType.I32,
            paramsType: [DataType.U8Array, DataType.U8Array],
            paramsValue: [pBuffer, pBufferLen],
        });
        if (errorCode === 0) {
            console.log('=====');
            console.log(pBuffer);
            const str = iconv.convert(pBuffer).toString();
            console.log(str);
            return str;
        }
    } catch (e) {
        console.error(e);
    }
    console.warn(errorCode);
}

testFun();

```

## expected result

<Buffer a4 a4 a4 e5>
中文

## actual result

try 5 ~ 10 times

1. sometime is right:
 
![image](https://github.com/zhangyuang/node-ffi-rs/assets/49306125/02c3436b-10b7-47d6-8731-26fd7e338d76)

2. sometime is wrong:

![image](https://github.com/zhangyuang/node-ffi-rs/assets/49306125/b1822188-244c-4bbe-b7bf-fabdc776ebbb)

## summary

1. Why are the results sometimes correct and sometimes wrong?
2. Not sure if it is caused by NodeJS Buffer problem or Rust conversion problem?
3. How to fix this?

**Comment by zhangyuang** [comment]
you should not use u8array to describe int* type

**Comment by zhangyuang** [comment]
use createPointer to create a pointer pointer to a number

**Comment by gn0262487838** [comment]
any sample code?

**Comment by zhangyuang** [comment]
you can read readme about createPointer or other issue，i dont have computer right now

**Comment by gn0262487838** [comment]
sometime i need to update int* length，so i fix code:

![image](https://github.com/zhangyuang/node-ffi-rs/assets/49306125/4bdf7e67-2b6c-4ada-9243-31c5d087779c)

but is not work and no show any error and result.

and then i change code:

![image](https://github.com/zhangyuang/node-ffi-rs/assets/49306125/c253d4a1-8c6c-457e-9abe-c1090c7924cd)

it still sometime right and sometime fail.

In my usage scenario, Sometimes need to change the int* length. It may be that a few successful case were returned.

**Comment by zhangyuang** [comment]
const len=createPointer（｛paramsType:[DataType.I32],paramsValue
:[4]
Pass the result[0] to ffi function with DataType.External and don’t need use unwrapPointer in this case

**Comment by gn0262487838** [comment]
![image](https://github.com/zhangyuang/node-ffi-rs/assets/49306125/d0f8d911-525b-4767-abfc-4edd3d2a63aa)

result: 
![image](https://github.com/zhangyuang/node-ffi-rs/assets/49306125/9bf68417-1277-460e-bb30-a9df21b4f6af)

it seems dont work for me...

**Comment by zhangyuang** [comment]
Oh, I find the problem is ffi-rs will release call params memory after one ffi call to avoid memory leak. but it's not necessary to free jsBuffer type memory because their memory will be managed by Node.js.

I will fix it in the next version

**Comment by zhangyuang** [comment]
The problem has been fixed in `ffi-rs@1.0.82`

**Comment by gn0262487838** [comment]
Thank u~it work for me.
